flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java

```diff
@@ -371,7 +374,14 @@ protected void addToSelectList(
         final SqlBasicCall call = (SqlBasicCall) node;
         final SqlOperator operator = call.getOperator();
 
-        if (operator instanceof SqlWindowTableFunction) {
+        if (node instanceof SqlExplicitModelCall) {
+            // Convert it so that model can be accessed in planner. SqlExplicitModelCall
+            // from parser can't access model.
+            SqlExplicitModelCall modelCall = (SqlExplicitModelCall) node;
+            return new SqlModelCall(modelCall);
+        }
+
+        if (operator instanceof SqlWindowTableFunction || operator instanceof SqlMLTableFunction) {
```

Comment:
Can we add a case to verify the change here?

Reply:
If it is not easy to add tests here, I perfer to add a TODO here and finish this part later.

Reply:
Sure. Will add a test to `flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/stream/sql/ColumnExpansionTest.java` after working E2E. Created https://issues.apache.org/jira/browse/FLINK-37819